src/components/AccordionSection/index.js

```diff
@@ -0,0 +1,176 @@
+        const { name, activeNames } = this.props;
+        const nameToToggle = name || this.name;
+
+        if (activeNames === undefined) {
+            return [nameToToggle];
+        }
+        if (isInArray(activeNames, nameToToggle)) {
+            return activeNames;
+        }
+        return activeNames.concat([nameToToggle]);
+    }
+
+    resolveNames() {
+        const { name, multiple } = this.props;
+        const nameToToggle = name || this.name;
+        if (multiple) {
+            return this.resolveNamesWhenMultiple();
+        }
+        return nameToToggle;
+    }
+
+    handleSelect(event) {
+        const { disabled, privateOnToggleSection } = this.props;
+        if (!disabled) {
+            privateOnToggleSection(event, this.resolveNames());
+        }
+    }
+
+    isExpanded() {
+        const { activeNames, name, multiple } = this.props;
+        const currentName = name || this.name;
+        if (multiple && Array.isArray(activeNames)) {
+            return isInArray(activeNames, currentName);
+        }
+        return activeNames === currentName;
+    }
+
+    render() {
+        const {
+            style,
+            disabled,
+            children,
+            label,
+            icon,
+            assistiveText,
+        } = this.props;
+
+        return (
+            <li className={this.getContainerClassNames()} style={style} disabled={disabled}>
+                <section>
+                    <div className="rainbow-accordion_summary">
+                        <h3 className="rainbow-accordion_summary-heading">
+                            <button
+                                className="rainbow-accordion_summary-button-heading"
+                                tabIndex={this.getTabIndex()}
+                                aria-controls={this.accordionDetailsId}
+                                aria-expanded={this.isSelected}
+                                onClick={this.handleSelect}>
+                                <RenderIf isTrue={!!icon}>
+                                    <span className="rainbow-accordion_summary-icon" aria-hidden="true">
+                                        {icon}
+                                    </span>
+                                </RenderIf>
+                                <RenderIf isTrue={!!label}>
+                                    <span title="Accordion Label">
+                                        {label}
+                                    </span>
+                                </RenderIf>
+                            </button>
+                        </h3>
+
+                        <ButtonIcon
+                            className="rainbow-accordion_summary-button"
+                            size="x-small"
+                            disabled={disabled}
+                            onClick={this.handleSelect}
+                            assistiveText={assistiveText}
+                            aria-haspopup="true"
```

Comment:
this is wrong aria-haspopup is an native attribute and ButtonIcon use a prop named ariaHaspopup and also you don't need to set it to "true", only put ariaHaspopup. In react an attribute without value means true so never use attribute="true" instead only put attribute.